pinot-core/src/main/java/org/apache/pinot/core/operator/query/NonScanBasedAggregationOperator.java

```diff
@@ -153,26 +154,30 @@ protected AggregationResultsBlock getNextBlock() {
   private static Double getMinValue(DataSource dataSource) {
     Dictionary dictionary = dataSource.getDictionary();
     if (dictionary != null) {
-      return toDouble(dictionary.getMinVal());
+      return toDouble(dictionary.getMinVal(), dataSource);
     }
-    return toDouble(dataSource.getDataSourceMetadata().getMinValue());
+    return toDouble(dataSource.getDataSourceMetadata().getMinValue(), dataSource);
   }
 
   private static Double getMaxValue(DataSource dataSource) {
     Dictionary dictionary = dataSource.getDictionary();
     if (dictionary != null) {
-      return toDouble(dictionary.getMaxVal());
+      return toDouble(dictionary.getMaxVal(), dataSource);
     }
-    return toDouble(dataSource.getDataSourceMetadata().getMaxValue());
+    return toDouble(dataSource.getDataSourceMetadata().getMaxValue(), dataSource);
   }
 
-  private static Double toDouble(Comparable<?> value) {
-    if (value instanceof Double) {
-      return (Double) value;
-    } else if (value instanceof Number) {
-      return ((Number) value).doubleValue();
-    } else {
-      return Double.parseDouble(value.toString());
+  private static Double toDouble(Comparable<?> value, DataSource dataSource) {
```

Comment:
We could just pass in the column name directly here I suppose?

Reply:
I guess so but I'm not sure how sensitive the place and tried to minimize potential overhead.